front/components/agent_builder/capabilities/shared/ProcessingMethodSection.tsx

```diff
@@ -38,52 +50,81 @@ export function ProcessingMethodSection() {
   });
   const sources = useWatch<CapabilityFormData, "sources">({ name: "sources" });
 
-  const { hasOnlyTablesSelected, hasSomeTablesSelected } = useMemo(() => {
-    if (!sources?.in?.length) {
-      return { hasOnlyTablesSelected: false, hasSomeTablesSelected: false };
-    }
+  const dataWarehouseServer = useMemo(
```

Comment:
`nit`: doesn't hurt but I guess we could get rid of the memoization here no? `mcpServerViewsWithKnowledge` should be quite stable

Reply:
it was less for `mcpServerViewsWithKnowledge` and more to avoid redoing that part when `sources.in` changes. But because we have to navigate back and forth to update it, it would be re-render regardless, so you're right, we can discard that.